File: jotupix_demo_windows/example/device/JDeviceManager.cpp
#include "JDeviceManager.h"

void JDeviceManager::CreateDevice(IJSend* pfnSender)
{
    std::shared_ptr<JotuPix> dev = std::make_shared<JotuPix>();
    dev->Init(pfnSender);

    m_psDevice = dev;
}

void JDeviceManager::RemoveDevice()
{
    m_psDevice = nullptr;
}

std::shared_ptr<JotuPix> JDeviceManager::GetDevice()
{
    return m_psDevice;
}

void JDeviceManager::SetDeviceInfoCache(JInfo info)
{
    m_sDevInfo = info;
}

JInfo& JDeviceManager::GetDeviceInfoCache()
{
    return m_sDevInfo;
}

File: jotupix_demo_ios/JotupixDemo/Demo/Device/JDeviceManager.m
#import "JDeviceManager.h"
#import "JotuPix.h"
#import "JInfo.h"
#import "JProtocol.h"

@interface JDeviceManager ()
{
    JotuPix *_device;
    JInfo *_devInfoCache;
}
@end

@implementation JDeviceManager

+ (instancetype)sharedInstance {
    static JDeviceManager *instance = nil;
    static dispatch_once_t onceToken;
    dispatch_once(&onceToken, ^{
        instance = [[JDeviceManager alloc] initPrivate];
    });
    return instance;
}

- (instancetype)initPrivate {
    self = [super init];
    if (self) {

    }
    return self;
}

#pragma mark - Methods

- (void)createDevice:(id<IJSend>)sender {
    JotuPix *dev = [[JotuPix alloc] init];
    [dev init:sender];  
    _device = dev;
}

- (void)removeDevice {
    _device = nil;
}

- (JotuPix *)getDevice {
    return _device;
}

- (void)setDeviceInfoCache:(JInfo *)info {
    _devInfoCache = info;
}

- (JInfo *)getDeviceInfoCache {
    return _devInfoCache;
}

@end
